sharedbuffers/mapped_struct.py

```diff
@@ -4165,119 +4173,79 @@ def map_file(cls, fileobj, offset = 0, size = None):
-        hi = length
-        lo = 0
 
-        if hkey < cython.cast(cython.p_ulonglong, pindex)[0]:
-            if check_equal:
-                return hi
-            else:
-                return lo
-        elif hkey > cython.cast(cython.p_ulonglong, pindex + stride0 * (hi-1))[0]:
-            return hi
+    numeric_A = cython.fused_type(
+        cython.char,
+        cython.uchar,
+        cython.short,
+        cython.ushort,
+        cython.int,
+        cython.uint,
+        cython.long,
+        cython.ulong,
+        cython.longlong,
+        cython.ulonglong,
+        cython.float,
+        cython.double,
+    )
 
-        if lo < hi:
-            # First iteration a quick guess assuming uniform distribution of keys
-            mid = min(hint, hi-1)
-            mkey = cython.cast(cython.p_ulonglong, pindex + stride0 * mid)[0]
-            if mkey < hkey:
-                # Got a lo guess, now skip-search forward for a hi
-                lo = mid = mid+1
-                skip = 32
-                while skip > 0 and mid + skip < hi:
-                    if cython.cast(cython.p_ulonglong, pindex + stride0 * (mid+skip))[0] < hkey:
-                        lo = mid+1
-                        mid += skip
-                        skip *= 2
-                    else:
-                        hi = mid + skip
-                        break
-            elif mkey > hkey:
-                # Got a hi guess, now skip-search backwards for a lo
-                hi = mid
-                skip = 32
-                while skip > 0 and mid > lo + skip:
-                    if cython.cast(cython.p_ulonglong, pindex + stride0 * (mid-skip))[0] > hkey:
-                        hi = mid
-                        mid -= skip
-                        skip *= 2
-                    else:
-                        lo = mid - skip
-                        break
-            else:
-                # hit, but must find the first
-                # good idea to go sequential because we assume collisions are unlikely
-                while mid > lo and cython.cast(cython.p_ulonglong, pindex + stride0 * (mid-1))[0] == hkey:
-                    mid -= 1
-                return mid
-        # Final stretch: search the remaining range with a regular binary search
-        while lo < hi:
-            mid = (lo+hi)//2
-            mkey = cython.cast(cython.p_ulonglong, pindex + stride0 * mid)[0]
-            if mkey < hkey:
-                lo = mid+1
-            elif mkey > hkey:
-                hi = mid
-            else:
-                while mid > lo and cython.cast(cython.p_ulonglong, pindex + stride0 * (mid-1))[0] == hkey:
-                    mid -= 1
-                return mid
-        # Check equality if requested
-        if check_equal and lo < length and cython.cast(cython.p_ulonglong, pindex + stride0 * lo)[0] != hkey:
-            lo = length
-        return lo
+    numeric_B = cython.fused_type(
```

Comment:
I understand that this happens because of the limitations on Cython when using Fused Types. I would add some comment about that in the code to make it explicit

Reply:
Sure.